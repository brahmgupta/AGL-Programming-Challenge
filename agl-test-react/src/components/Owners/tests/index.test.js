import React from 'react';
import { render } from '@testing-library/react';
import OwnerList from '../index';

const mockOwners = [{"ownerGender":"Male","catNames":["Garfield","Jim","Max","Tom"]}];

describe("<OwnerList />", () => {
  it("renders the component", () => {
    const container = render(<OwnerList ownerWithPets={mockOwners} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render All owners gender", () => {
    const { queryAllByTestId } = render(<OwnerList ownerWithPets={mockOwners} />);
    expect(queryAllByTestId("owner").length).toBe(mockOwners.length);
  });

  it("should NOT render any pet", () => {
    const { queryAllByTestId } = render(<OwnerList />);
    expect(queryAllByTestId("owner").length).toBe(0);
  });
});
