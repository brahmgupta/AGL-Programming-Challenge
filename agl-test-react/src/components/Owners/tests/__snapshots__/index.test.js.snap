// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OwnerList /> renders the component 1`] = `undefined`;
